polarCoordinateTraversal_dotprod.m

```diff
@@ -0,0 +1,198 @@
+r = delta_radius;
+while ((ray_circle_vector(1)^2 + ray_circle_vector(2)^2 > r^2) && ...
+        r < circle_max_radius)
+    r = r + delta_radius        ;
+end
+% Find the intersection times for the ray and the radial shell containing the ray
+% at t_begin
+ray_unit_vector = 1/sqrt(ray_direction(1)^2 + ray_direction(2)^2)...
+    .* [ray_direction(1);  ray_direction(2)]';
+v = dot(ray_circle_vector,ray_unit_vector);
+discr = r^2 - (dot(ray_circle_vector,ray_circle_vector) - v^2);
+d = sqrt(discr);
+p1 = ray_origin + (v-d).*ray_unit_vector;
+p2 = ray_origin + (v+d).*ray_unit_vector;
+t1 = (p1(1)-ray_origin(1))/ray_direction(1);
+t2 = (p2(1)-ray_origin(2))/ray_direction(2);
+% The ray may not intersect the grid
+% at all; in particular if the ray is outside the grid at t_begin.
+if (t1 < t_begin && t2 < t_begin )
+fprintf("\nray doesn't intersect polar grid for t_begin!")
+return;
+end
+% If there is a ray/shell intersection, then set the radial voxel ID. 
+current_voxel_ID_r = 1 + (circle_max_radius - r)/delta_radius
+if verbose
+    text(p1(1), p1(2), 'POI_r');
+    fprintf('RADIAL HIT INIT.\n');
+end
+
+% II. Calculate Voxel ID Theta.
+current_voxel_ID_theta = floor(atan2(ray_start_y - circle_center_y, ray_start_x - circle_center_x) * num_angular_sections / (2 * pi));
+if current_voxel_ID_theta < 0
+    current_voxel_ID_theta = num_angular_sections + current_voxel_ID_theta;
+end
+angular_voxels = [current_voxel_ID_theta];
+radial_voxels = [current_voxel_ID_r];
+
+% III. TRAVERSAL PHASE
+tol = 10^-6;
+t = t_begin; 
+while (t < t_end)
+    % 1. Calculate tMaxR, tMaxTheta
+    [tMaxR] = radial_hit_dotprod(ray_origin, ray_direction, ...
+        current_voxel_ID_r, circle_center, circle_max_radius, delta_radius, t, verbose);
+    pause
+    [tMaxTheta, tStepTheta] = angular_hit(ray_origin, ray_direction, current_voxel_ID_theta,...
+        num_angular_sections, circle_center, t, verbose);
+    pause
+    % 2. Compare tMaxTheta, tMaxR
+    if (tMaxTheta < tMaxR)
+        t = tMaxTheta;
+        current_voxel_ID_theta = current_voxel_ID_theta + tStepTheta;
+        if verbose
+            new_x_position = ray_origin(1) + ray_direction(1) * tMaxTheta;
+            new_y_position = ray_origin(2) + ray_direction(2) * tMaxTheta;
+            text(new_x_position, new_y_position, 'POI_t');
+            fprintf('ANGULAR HIT.\n');
+        end
+    else
+        t = tMaxR;
+        p = ray_origin + t.*ray_direction;
+        r_new = sqrt((p(1) - circle_center(1))^2 + (p(2) - circle_center(2))^2);
+        if (abs(r_new - r) < tol)
+            tStepR = 0; 
+            current_voxel_ID_r = current_voxel_ID_r; 
+        elseif (r_new - r > 0) 
+            tStepR = -1; 
+            current_voxel_ID_r = current_voxel_ID_r -1;
+        else
+            tStepR = 1; 
+            current_voxel_ID_r = current_voxel_ID_r + 1;
+        end
+        r = r_new;
+        new_x_position = ray_origin_x + ray_direction_x * tMaxR;
```

Comment:
Let's move these inside the ```if verbose ... end``` block since they aren't required in our actual calculation and are used for debugging

Reply:
Only the case for line 180.

Reply:
Yep just ```new_x_position``` and ```new_y_position```